lib/facility.rb

```diff
@@ -1,14 +1,65 @@
+require './lib/dmv'
+require './lib/facility'
+require './lib/vehicle'
+require './lib/dmv_data_service'
+
+
 class Facility
-  attr_reader :name, :address, :phone, :services
+  attr_reader :name, :address, :phone, :services, :registered_vehicles, :collected_fees
 
-  def initialize(name, address, phone)
-    @name = name
-    @address = address
-    @phone = phone
-    @services = []
+  def initialize(facility_details)
+    @address = facility_details[:address]
+    @collected_fees = 0
+    @name = facility_details[:name]
+    @phone = facility_details[:phone]
+    @registered_vehicles = []
+    @services = [] 
+    
   end
 
-  def add_services(service)
+  def add_service(service)
     @services << service
   end
+
+  
+  def register_vehicle(vehicle)
+    vehicle.change_plate
+    vehicle.change_date
+    @registered_vehicles << vehicle
+    if vehicle.antique?
+      @collected_fees += 25
+    elsif vehicle.electric_vehicle?
+      @collected_fees += 200
+    else
+      @collected_fees += 100
+    end
+    
+  end
+  
+  def administer_written_test(registrant)
+    if @services.include?('Written Test') == true && registrant.permit? ==true && registrant.age >=16
```

Comment:
Nice job with your boolean logic. One thing you can omit to make this cleaner, is if something already returns a boolean, we don't need to also do a comparison with a boolean. So line 40 could read `if @services.include?('Written Test') && registrant.permit? && registrant.age >=16`